### ISSUE TITLE
[Bug] Conflict when working in two TeX projects

### ISSUE THREAD

**Issue by jose-a-sa** [James-Yu/LaTeX-Workshop#95]
Imagine I have 2 multi .tex projects and Paper.tex and Thesis.tex both with `\begin{document}`. Then for exemple if have _Paper.tex_ and _ThesisChapter1.tex_ open and I tell vscode to do `latex-workshop.build` or `latex-workshop.tab` or `latex-workshop.clean` when I have _ThesisChapter1.tex_ focused in the editor, then it builds/shows/cleans the Paper.tex and not Thesis.tex . I have all tex files with the correspondent `% !TeX root = ../rootTeX.tex` magic command.

My workspace:
```
RootWorkspace
│  (some Mathematica .nb to export plot to both Figures/ folders )
│  Bibliography.bib
│
└───Thesis
│   │   Thesis.tex
│   │   ...
│   │
│   └───Chapters
│   │   │   ThesisChapter1.tex
│   │   │   ...
│   │   
│   └───Figures
│   │  ...
│
└───Paper
│   │   Paper.tex
│   │   ...
│   │
│   └───Sections
│   │   │   PaperSection1.tex
│   │   │   ...
│   │   
│   └───Figures
│   │  ...
```
Simply put, if I have only _ThesisChapter1.tex_ and _PaperSection1.tex_ open in the editor, if I do `latex-workshop.build` on either it will not build the one that it's focused on the editor but, I believe, the one that was build last.

**Comment by jose-a-sa** [comment]
@jabooth @James-Yu 
**Update:** also when both Paper.pdf and both Thesis.pdf are open in tab mode synctex PDF->TEX work perfectly 👍, but TEX->PDF only Works if the .tex file matches the .pdf, when it should pop out the correspondent tab.

Regards, José

**Comment by James-Yu** [comment]
Can you please provide LaTeX Workshop log on you changing active editor from thesis.tex to paper.tex and vice cersa? That will help identifying the problem. In fact, on changing active editor the extension should be able to check the new root file. And that works quite well on my side.

As to tex->pdf synctex, I will check if there's way to activate the corresponding tab.

**Comment by James-Yu** [comment]
I've checked the second issue and found out that it is not possible.

VS code internally close the pdf files if the tab is blurred (inactive). So it will not receive any instructions, nor does the extension know if the pdf should be opened or not.

**Comment by James-Yu** [comment]
For your first problem, the log on my side looks like
```
[10:15:01] Root file remains unchanged from: \path\to\main.tex.
[10:15:01] Found root file from active editor: \path\to\r1\response.tex
[10:15:01] Root file changed from: \path\to\main.tex. Find all dependencies.
[10:15:01] Root file changed -> cleaning up old file watcher.
[10:15:01] Instatiating new file watcher for \path\to\r1\response.tex
[10:15:01] Parsing \path\to\r1\response.tex
```
You can clearly see the root file changed from one tex to another. My folder does not have any magic commands set.

**Comment by jose-a-sa** [comment]
I opened vscode with _ThesisChapter1.tex_ and _PaperSection1.tex_ and tried to use `latex-workshop.clean` and it gives na error (cant find root).
Next I removed the magic comment from both files and saved, then vscode tries to build both files and fails. Here's the log output: 
````
[04:58:41] Initializing LaTeX Workshop.
[04:58:41] Creating LaTeX Workshop http and websocket server.
[04:58:41] LaTeX Workshop initialized.
[04:58:41] Linter for root file started.
[04:58:41] Server created on 127.0.0.1:56886
[04:58:41] Default environments loaded
[04:58:41] Default commands loaded
[04:58:41] Default unimathsymbols loaded
[04:59:51] Linter for active file started.
[04:59:51] Linter for active file running command chktex with arguments -wall,-n1,-n3,-n8,-n22,-n30,-n44,-e16,-q,-I0,-f%f:%l:%c:%d:%k:%n:%m

[04:59:51] Linter for active file started.
[04:59:51] Linter for active file running command chktex with arguments -wall,-n1,-n3,-n8,-n22,-n30,-n44,-e16,-q,-I0,-f%f:%l:%c:%d:%k:%n:%m

[04:59:51] Linter for active file failed with exit code null and error:
  
[04:59:51] Linter for active file successfully finished in 0s 350ms
[04:59:51] Linter log parsed with 1 messages.
[04:59:52] Linter for root file started.
[04:59:52] BUILD command invoked.
[04:59:52] Cannot find LaTeX root file.
[04:59:56] Linter for active file started.
[04:59:56] Linter for active file running command chktex with arguments -wall,-n1,-n3,-n8,-n22,-n30,-n44,-e16,-q,-I0,-f%f:%l:%c:%d:%k:%n:%m

[04:59:57] Linter for active file successfully finished in 0s 487ms
[04:59:57] Linter log parsed with 1 messages.
[05:00:04] Linter for active file started.
[05:00:04] Linter for active file running command chktex with arguments -wall,-n1,-n3,-n8,-n22,-n30,-n44,-e16,-q,-I0,-f%f:%l:%c:%d:%k:%n:%m

[05:00:04] Linter for active file successfully finished in 0s 428ms
[05:00:04] Linter log parsed with 1 messages.
[05:00:04] Linter for active file started.
[05:00:04] Linter for active file running command chktex with arguments -wall,-n1,-n3,-n8,-n22,-n30,-n44,-e16,-q,-I0,-f%f:%l:%c:%d:%k:%n:%m

[05:00:05] Linter for active file successfully finished in 0s 377ms
[05:00:05] Linter log parsed with 1 messages.
[05:00:05] Linter for root file started.
[05:00:05] BUILD command invoked.
[05:00:05] Cannot find LaTeX root file.
[05:00:08] CLEAN command invoked.
[05:00:14] CLEAN command invoked.
[05:00:19] Linter for active file started.
[05:00:19] Linter for active file running command chktex with arguments -wall,-n1,-n3,-n8,-n22,-n30,-n44,-e16,-q,-I0,-f%f:%l:%c:%d:%k:%n:%m

[05:00:20] Linter for active file started.
[05:00:20] Linter for active file running command chktex with arguments -wall,-n1,-n3,-n8,-n22,-n30,-n44,-e16,-q,-I0,-f%f:%l:%c:%d:%k:%n:%m

[05:00:20] Linter for active file failed with exit code null and error:
  
[05:00:20] Linter for active file successfully finished in 0s 363ms
[05:00:20] Linter log parsed with 1 messages.
[05:00:20] Linter for active file started.
[05:00:20] Linter for active file running command chktex with arguments -wall,-n1,-n3,-n8,-n22,-n30,-n44,-e16,-q,-I0,-f%f:%l:%c:%d:%k:%n:%m

[05:00:21] Linter for active file successfully finished in 0s 413ms
[05:00:21] Linter log parsed with 1 messages.
[05:00:21] Linter for root file started.
[05:00:21] BUILD command invoked.
[05:00:21] Cannot find LaTeX root file.
[05:00:23] CLEAN command invoked.
[05:00:34] Linter for active file started.
[05:00:34] Linter for active file running command chktex with arguments -wall,-n1,-n3,-n8,-n22,-n30,-n44,-e16,-q,-I0,-f%f:%l:%c:%d:%k:%n:%m

[05:00:35] Linter for active file successfully finished in 0s 366ms
[05:00:35] Linter log parsed with 1 messages.
[05:00:36] Linter for active file started.
[05:00:36] Linter for active file running command chktex with arguments -wall,-n1,-n3,-n8,-n22,-n30,-n44,-e16,-q,-I0,-f%f:%l:%c:%d:%k:%n:%m

[05:00:36] Linter for active file successfully finished in 0s 424ms
[05:00:36] Linter log parsed with 1 messages.
[05:01:10] Linter for active file started.
[05:01:10] Linter for active file running command chktex with arguments -wall,-n1,-n3,-n8,-n22,-n30,-n44,-e16,-q,-I0,-f%f:%l:%c:%d:%k:%n:%m

[05:01:10] Linter for active file successfully finished in 0s 438ms
[05:01:10] Linter log parsed with 1 messages.
[05:01:11] CLEAN command invoked.
[05:03:53] Linter for active file started.
[05:03:53] Linter for active file running command chktex with arguments -wall,-n1,-n3,-n8,-n22,-n30,-n44,-e16,-q,-I0,-f%f:%l:%c:%d:%k:%n:%m

[05:03:53] Linter for active file successfully finished in 0s 472ms
[05:03:53] Linter log parsed with 1 messages.
[05:04:29] Linter for active file started.
[05:04:29] Linter for active file running command chktex with arguments -wall,-n1,-n3,-n8,-n22,-n30,-n44,-e16,-q,-I0,-f%f:%l:%c:%d:%k:%n:%m

[05:04:29] Linter for active file started.
[05:04:29] Linter for active file running command chktex with arguments -wall,-n1,-n3,-n8,-n22,-n30,-n44,-e16,-q,-I0,-f%f:%l:%c:%d:%k:%n:%m

[05:04:29] Linter for active file failed with exit code null and error:
  
[05:04:30] Linter for active file successfully finished in 0s 448ms
[05:04:30] Linter log parsed with 1 messages.
[05:04:35] Linter for active file started.
[05:04:35] Linter for active file running command chktex with arguments -wall,-n1,-n3,-n8,-n22,-n30,-n44,-e16,-q,-I0,-f%f:%l:%c:%d:%k:%n:%m

[05:04:36] Linter for active file started.
[05:04:36] Linter for active file running command chktex with arguments -wall,-n1,-n3,-n8,-n22,-n30,-n44,-e16,-q,-I0,-f%f:%l:%c:%d:%k:%n:%m

[05:04:36] Linter for active file failed with exit code null and error:
  
[05:04:36] Linter for active file successfully finished in 0s 395ms
[05:04:36] Linter log parsed with 1 messages.
[05:04:37] Linter for root file started.
[05:04:37] BUILD command invoked.
[05:04:37] Cannot find LaTeX root file.
[05:04:42] Linter for active file started.
[05:04:42] Linter for active file running command chktex with arguments -wall,-n1,-n3,-n8,-n22,-n30,-n44,-e16,-q,-I0,-f%f:%l:%c:%d:%k:%n:%m

[05:04:43] Linter for active file started.
[05:04:43] Linter for active file running command chktex with arguments -wall,-n1,-n3,-n8,-n22,-n30,-n44,-e16,-q,-I0,-f%f:%l:%c:%d:%k:%n:%m

[05:04:43] Linter for active file failed with exit code null and error:
  
[05:04:43] Linter for active file started.
[05:04:43] Linter for active file running command chktex with arguments -wall,-n1,-n3,-n8,-n22,-n30,-n44,-e16,-q,-I0,-f%f:%l:%c:%d:%k:%n:%m

[05:04:43] Linter for active file failed with exit code null and error:
  
[05:04:43] Linter for active file successfully finished in 0s 379ms
[05:04:43] Linter log parsed with 1 messages.
[05:04:46] Linter for active file started.
[05:04:46] Linter for active file running command chktex with arguments -wall,-n1,-n3,-n8,-n22,-n30,-n44,-e16,-q,-I0,-f%f:%l:%c:%d:%k:%n:%m

[05:04:46] Linter for active file successfully finished in 0s 439ms
[05:04:46] Linter log parsed with 1 messages.
[05:04:48] Linter for active file started.
[05:04:48] Linter for active file running command chktex with arguments -wall,-n1,-n3,-n8,-n22,-n30,-n44,-e16,-q,-I0,-f%f:%l:%c:%d:%k:%n:%m

[05:04:48] Linter for active file started.
[05:04:48] Linter for active file running command chktex with arguments -wall,-n1,-n3,-n8,-n22,-n30,-n44,-e16,-q,-I0,-f%f:%l:%c:%d:%k:%n:%m

[05:04:48] Linter for active file failed with exit code null and error:
  
[05:04:48] Linter for active file started.
[05:04:48] Linter for active file running command chktex with arguments -wall,-n1,-n3,-n8,-n22,-n30,-n44,-e16,-q,-I0,-f%f:%l:%c:%d:%k:%n:%m

[05:04:48] Linter for active file failed with exit code null and error:
  
[05:04:49] Linter for active file successfully finished in 0s 377ms
[05:04:49] Linter log parsed with 1 messages.
[05:04:52] Linter for root file started.
[05:04:52] BUILD command invoked.
[05:04:52] Cannot find LaTeX root file.
[05:04:54] Linter for active file started.
[05:04:54] Linter for active file running command chktex with arguments -wall,-n1,-n3,-n8,-n22,-n30,-n44,-e16,-q,-I0,-f%f:%l:%c:%d:%k:%n:%m

[05:04:55] Linter for active file successfully finished in 0s 506ms
[05:04:55] Linter log parsed with 1 messages.
[05:04:55] Linter for root file started.
[05:04:55] BUILD command invoked.
[05:04:55] Cannot find LaTeX root file.
[05:05:06] Linter for active file started.
[05:05:06] Linter for active file running command chktex with arguments -wall,-n1,-n3,-n8,-n22,-n30,-n44,-e16,-q,-I0,-f%f:%l:%c:%d:%k:%n:%m

[05:05:06] Linter for active file successfully finished in 0s 402ms
[05:05:06] Linter log parsed with 1 messages.
````

**Comment by jose-a-sa** [comment]
**Remenber:** both projects root files are not in the `workspaceRoot`

**Comment by James-Yu** [comment]
Sorry you log is too noisy. What's the NEW log entries when you switch from one file to another?

**Comment by James-Yu** [comment]
A minimal working example shared by gist is always welcomed

**Comment by jose-a-sa** [comment]
I dont know how to use gist ...

**Comment by jose-a-sa** [comment]
When I change between file I only get linter logs

**Comment by James-Yu** [comment]
Do you have any error messages in developer tools? `ctrl`+`shift`+`i`.

tbh the issue is weird. Neither can I reproduce the problem nor it was reported by others. I suspect it might be a local problem.

**Comment by James-Yu** [comment]
By the way, what if you switch between two `tex` files with `\begin{document}` commands?

**Comment by jose-a-sa** [comment]
The thing is if I open for exemple Paper.tex I get all the correct logs
`05:22:19] Found root file from active editor: (...)\Dissertation\Paper\Paper.tex`
also we I open main.tex (thesis)
````
[05:23:41] Found root file from active editor: (...)\Dissertation\Thesis\main.tex
[05:23:41] Root file changed from: (...)\Dissertation\Paper\Paper.tex. Find all dependencies.
[05:23:41] Root file changed -> cleaning up old file watcher.
[05:23:41] Instatiating new file watcher for (...)\Dissertation\Thesis\main.tex
[05:23:41] Parsing (...)\Dissertation\Thesis\main.tex
[05:23:41] Adding (...)\Dissertation\Thesis\Chapters\Chapter1.tex to file watcher.
[05:23:41] Parsing (...)\Dissertation\Thesis\Chapters\Chapter1.tex
[05:23:41] Adding (...)\Dissertation\Thesis\Chapters\Chapter2.tex to file watcher.
[05:23:41] Parsing (...)\Dissertation\Thesis\Chapters\Chapter2.tex
[05:23:42] Adding (...)\Dissertation\Thesis\Appendices\AppendixA.tex to file watcher.
[05:23:42] Parsing (...)\Dissertation\Thesis\Appendices\AppendixA.tex
[05:23:42] Found .bib file (...)\Dissertation\Bibliography.bib
[05:23:42] .bib file (...)\Dissertation\Bibliography.bib is already being watched.
```
After I close both root files I get always the same log when I change between content .tex files
`[05:26:04] Root file remains unchanged from: d:\Jose\MEGA\University\5th YEAR\Dissertation\Thesis\main.tex`

**Comment by James-Yu** [comment]
It seems there's some problem when detecting your magic command. Can you please post the comments with your personal information removed?

**Comment by James-Yu** [comment]
The comment is `TEX` instead of `TeX`. Please correct and check again.

**Comment by jose-a-sa** [comment]
If you switch between two tex files with \begin{document} commands, the `latex-workshop` identifies all file dependencies in each .tex correctly. The problem is: 
I have to change to the the specific root tex file so that vscode identifies which project I'm working on before changing to any chapter/section file

**Comment by jose-a-sa** [comment]
Then thereis a typo on the README.md file

**Comment by James-Yu** [comment]
Is it working correctly now?

**Comment by jose-a-sa** [comment]
Yep it work now @James-Yu, thank you very much. Still could you look into the TEX->PDF synctex situation ?
Would be nice if `latex-workshop` could identify these root .tex without the `%!TEX` magic command.

Regards, José

**Comment by James-Yu** [comment]
For your first issue, as I explained before, it is limited by vscode itself so nothing can be done from the extension side.

For the second one, it is not possible because that will involve a huge amount of disc IO to traverse all root folders and subfolders. In addition, it is extremely hard to identify the root file of a sub `tex` file. What if the same file is included in two roots?

**Comment by jose-a-sa** [comment]
depends on the magic command

**Comment by jose-a-sa** [comment]
what if `latex-workshop.synctex` would also do `latex-workshop.tab`

**Comment by James-Yu** [comment]
Then the browser-viewer users will be annoyed.
